packages/ws/src/Decorators/Message.ts

```diff
@@ -0,0 +1,78 @@
+import { BadRequestError, initializeMetadata, initializeMetadataMethod, MetadataResolver, ValidationProvider, ValidationTypes } from '@vercube/core';
+import { BaseDecorator, createDecorator, Inject, InjectOptional } from '@vercube/di';
+import { WebsocketService } from '../Services/WebsocketService';
+import { WebsocketServiceKey } from '../Utils/WebsocketServiceKey';
+
+interface MessageDecoratorOptions {
+  event: string;
+  validationSchema?: ValidationTypes.Schema;
+}
+
+class MessageDecorator extends BaseDecorator<MessageDecoratorOptions> {
+
+  @Inject(MetadataResolver)
+  private gMetadataResolver!: MetadataResolver;
+
+  @Inject(WebsocketServiceKey)
+  private gWebsocketService!: WebsocketService;
+
+  @InjectOptional(ValidationProvider)
+  private gValidationProvider: ValidationProvider | null;
+
+  public override created(): void {
+    initializeMetadata(this.prototype);
+    const method = initializeMetadataMethod(this.prototype, this.propertyName);
+
+    const namespace = this.gMetadataResolver.resolveNamespace({
+      instance: this.instance
+    });
+
+    const shouldBroadcast = method.args.find(e => e.type === 'broadcast');
+    const shouldBroadcastOthers = method.args.find(e => e.type === 'broadcast_others');
+    const shouldEmit = method.args.find(e => e.type === 'emit') && !shouldBroadcast;
+
+    this.gWebsocketService.registerMessageHandler(
```

Comment:
whole this code has to be part of WSS Service instead of decorator. Decorators should only use method from service instead of build entire logic itself. You can create more then 1 service if you need (for internal usage)